src/EditorFeatures/Core/SymbolSearch/SymbolSearchUpdateEngine.Update.cs

```diff
@@ -110,19 +104,19 @@ public Updater(SymbolSearchUpdateEngine service, string source, string localSett
                 }
 
                 // Keep on looping until we're told to shut down.
-                while (!_service._updateCancellationToken.IsCancellationRequested)
+                while (!cancellationToken.IsCancellationRequested)
                 {
-                    await _service.LogInfoAsync("Starting update").ConfigureAwait(false);
+                    await _service.LogInfoAsync("Starting update", cancellationToken).ConfigureAwait(false);
                     try
                     {
-                        var delayUntilNextUpdate = await UpdateDatabaseInBackgroundWorkerAsync().ConfigureAwait(false);
+                        var delayUntilNextUpdate = await UpdateDatabaseInBackgroundWorkerAsync(cancellationToken).ConfigureAwait(false);
 
-                        await _service.LogInfoAsync($"Waiting {delayUntilNextUpdate} until next update").ConfigureAwait(false);
-                        await Task.Delay(delayUntilNextUpdate, _service._updateCancellationToken).ConfigureAwait(false);
+                        await _service.LogInfoAsync($"Waiting {delayUntilNextUpdate} until next update", cancellationToken).ConfigureAwait(false);
+                        await Task.Delay(delayUntilNextUpdate, cancellationToken).ConfigureAwait(false);
                     }
                     catch (OperationCanceledException)
                     {
-                        await _service.LogInfoAsync("Update canceled. Ending update loop").ConfigureAwait(false);
+                        await _service.LogInfoAsync("Update canceled. Ending update loop", cancellationToken).ConfigureAwait(false);
```

Comment:
in new design, 
VS side. cancellation belong to each call. basically each call can have different cancellation now which is different than before.

Service side. StreamJson owns cancellation. it can raise the cancellation regardless of connection being closed or not. now service need to differentiate cancellation from StreamJson and connection being disconnected case. and decide how to deal with connection being disconnected case without cancellation raised from StreamJson.